spring-data-sample/src/main/java/sample/SampleApp.java

```diff
@@ -0,0 +1,37 @@
+package sample;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.ExitCodeGenerator;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import picocli.CommandLine;
+import picocli.CommandLine.IFactory;
+import sample.command.SampleCommand;
+
+@Configuration
+@ComponentScan
+@EnableAutoConfiguration
+public class SampleApp implements CommandLineRunner, ExitCodeGenerator {
+  private int exitCode;
+
+  @Autowired IFactory factory;
+
+  @Autowired SampleCommand sampleCommand;
+
+  @Override
+  public void run(String... args) {
+    exitCode = new CommandLine(sampleCommand, factory).execute(args);
+  }
+
+  @Override
+  public int getExitCode() {
+    return exitCode;
+  }
+
+  public static void main(String[] args) {
```

Comment:
The original application invokes `sample.command.SampleCommand`, but this application starts with this class as a Spring Boot application entry point.